src/Parsers/MySQL/ASTDDLTableIdentifier.cpp

```diff
@@ -0,0 +1,74 @@
+#include <Parsers/MySQL/ASTDDLTableIdentifier.h>
+
+#include <Interpreters/StorageID.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/ASTLiteral.h>
+
+
+namespace DB
+{
+
+namespace MySQLParser
+{
+
+ASTPtr ASTDDLTableIdentifier::clone() const
+{
+    auto res = std::make_shared<ASTDDLTableIdentifier>(*this);
+    res->children.clear();
+    return res;
+}
+
+bool ParseDDLTableIdentifier::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ASTPtr table;
+    if (ParserKeyword("CREATE TEMPORARY TABLE").ignore(pos, expected) || ParserKeyword("CREATE TABLE").ignore(pos, expected))
+    {
+        ParserKeyword("IF NOT EXISTS").ignore(pos, expected);
```

Comment:
It's a bit tricky that parser for TableIdentifier also parses other parts of query. And we don't actually need `ASTDDLTableIdentifier` for this, because it cannot be a part for some larger AST. Let's just make static function `tryExtractTableNameFromDDL` or something like that instead of `ParseDDLTableIdentifier::parseImpl`

Reply:
fixed